kafka-rest/src/main/java/io/confluent/kafkarest/exceptions/BadRequestException.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.exceptions;
+
+import javax.ws.rs.core.Response.Status;
+
+public class BadRequestException extends StatusCodeException {
+
+  public BadRequestException(String detail) {
+    this("Bad Request", detail);
+  }
+
+  public BadRequestException(String detail, Throwable cause) {
+    this("Bad Request", detail, cause);
+  }
+
+  public BadRequestException(Integer code, String detail) {
+    this(code, "Bad Request", detail);
+  }
+
+  public BadRequestException(int code, String detail, Throwable cause) {
+    this(code, "Bad Request", detail, cause);
+  }
```

Comment:
nit: Canonically _Bad Request_ is HTTP 400, so I wonder in which case would we need any of these ctors?

Reply:
This code is not the HTTP code, but rather an application error code that we return in the response body. Users can later search for it in our documentation to understand the error better.

Reply:
Two comments then:
1. Do we need these now though, if we are not using them anywhere?
2. If you think we need them, could you please add Javadoc for them explaining the application error bit?

Reply:
Done.